MaxText/multimodal_utils.py

```diff
@@ -389,9 +409,13 @@ def pre_process_image(image, model_name):
 def reformat_prompt(prompt, model_name):
   """Reformat prompt for different models."""
   if model_name in ["gemma3-4b", "gemma3-12b", "gemma3-27b"]:
+    if not GEMMA_IMAGE_PLACEHOLDER_IN_PROMPT in prompt:
+      prompt = GEMMA_IMAGE_PLACEHOLDER_IN_PROMPT + prompt
     formatted_prompt = f"<start_of_turn>user\n{prompt}<end_of_turn>\n<start_of_turn>model\n"
     return formatted_prompt
   elif model_name in ["llama4-17b-16e", "llama4-17b-128e"]:
+    if not LLAMA4_IMAGE_PLACEHOLDER_IN_PROMPT in prompt:
+      prompt = LLAMA4_IMAGE_PLACEHOLDER_IN_PROMPT + prompt
```

Comment:
could you document how user can choose custom placeholder in the prompt (e.g. in SFT pipeline).

Reply:
Good suggestion. I just added image_placeholder in base.yml to support custom placeholder.